### PR TITLE
refactor-documentation-for-signals

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**

### CODE REVIEW

File: src/docs/user/ProActiveUserGuide.adoc [hunk 1]
```diff
@@ -2621,8 +2621,12 @@ The *Signal API* is a high-level API that enables users to interact with running
...
+In summary and for simple use, just go to <<_proactive_studio, ProActive Studio>> and use the predefined templates in the _Controls_ bucket, _4. Signal Templates_ project: for example: *Check_For_Signals*, *Wait_For_Signals* or *Wait_For_Signal_With_Variables*.
 Fill up the Task Variable *SIGNALS*, and use those templates in your own workflow to be able to receive input from, for instance, the Workflow Execution portal or Scheduler portal.
+*Check_For_Signals* is a template workflow that sends a ready notification for all the signals specified in the variable SIGNALS, then loops until one signal among those specified is received by the job.
+*Wait_For_Signals* also sends a ready notification for all the signals specified in the variable SIGNALS, loops until one signal among those specified is received by the job and then removes the signal.
+*Wait_For_Signal_With_Variables* is template workflow that waits until one signal (with input parameters) that is specified in the variable SIGNAL is added to the set of job signals.
```
Comment (nebilbenmabrouk) [inline src/docs/user/ProActiveUserGuide.adoc:2628]: is **a** template workflow

Comment (nebilbenmabrouk) [inline src/docs/user/ProActiveUserGuide.adoc:2628]: I suggest to add few words for each workflow template to explain its main difference compared to the others, and (possibly) in  which case to use it

**Review by AlexFalessi — APPROVED** [review]

File: src/docs/user/ProActiveUserGuide.adoc [hunk 2]
```diff
@@ -2621,8 +2621,15 @@ The *Signal API* is a high-level API that enables users to interact with running
...
 You can try the <<_examples_2,examples>> below to better understand how it works.
 
-In summary and for simple use, just go to <<_proactive_studio, ProActive Studio>> and use the 2 predefined templates in the Controls menu: *Check_For_Signals* or *Wait_For_Signals*.
+In summary and for simple use, just go to <<_proactive_studio, ProActive Studio>> and use the predefined templates in the _Controls_ bucket, _4. Signal Templates_ project: for example: *Check_For_Signals*, *Wait_For_Signals* or *Wait_For_Signal_With_Variables*.
 Fill up the Task Variable *SIGNALS*, and use those templates in your own workflow to be able to receive input from, for instance, the Workflow Execution portal or Scheduler portal.
+*Check_For_Signals* is a template workflow that sends a ready notification for all the signals specified in the variable SIGNALS, then loops until one signal among those specified is received by the job.
+This workflow is composed by two tasks. The first sends a ready notification for all of the signals. The second task checks if the signals are received and triggers the first task in a loop untill all of the signals are received.
```
Comment (aminelouati) [inline src/docs/user/ProActiveUserGuide.adoc:2627]: composed by -> composed of

Comment (aminelouati) [reply src/docs/user/ProActiveUserGuide.adoc:2627]: The first --> The first task

Comment (aminelouati) [reply src/docs/user/ProActiveUserGuide.adoc:2627]: untill -> until

Comment (aminelouati) [reply src/docs/user/ProActiveUserGuide.adoc:2627]: untill all of the signals are received. -> can you check this sentence please as you said in the sentence before "until one signal among those specified is received by the job"

File: src/docs/user/ProActiveUserGuide.adoc [hunk 3]
```diff
@@ -2621,8 +2621,15 @@ The *Signal API* is a high-level API that enables users to interact with running
...
+*Wait_For_Signals* also sends a ready notification for all the signals specified in the variable SIGNALS, loops until one signal among those specified is received by the job and then removes the signal.
```
Comment (aminelouati) [inline src/docs/user/ProActiveUserGuide.adoc:2628]: can you double check the description with respect to the previous workflow description ?

File: src/docs/user/ProActiveUserGuide.adoc [hunk 4]
```diff
@@ -2621,8 +2621,15 @@ The *Signal API* is a high-level API that enables users to interact with running
...
+Unlike _Check_For_Signals_, _Wait_For_Signals_ workflow contains a single task that sends a ready notification for all of the signals and then checks if the signals are received. This task is executed in loop untill all of the signals are received.
```
Comment (aminelouati) [inline src/docs/user/ProActiveUserGuide.adoc:2629]: untill --> until (please make sure to fix it everywhere)

File: src/docs/user/ProActiveUserGuide.adoc (hunk 2 above)
Comment (aminelouati) [reply src/docs/user/ProActiveUserGuide.adoc:2627]: a ready notification for all of the signals -> a ready notification for all signals

File: src/docs/user/ProActiveUserGuide.adoc [hunk 5]
```diff
@@ -2621,9 +2621,20 @@ The *Signal API* is a high-level API that enables users to interact with running
...
+
+*Wait_For_Signals* is a template workflow that contains a single task, which sends a ready notification for the signals specified in the variable SIGNALS, then checks if one or more signals are received by the job. If not, the workflow iterates in a loop where each iteration lasts one minute.
+This workflow allows users to iteratively wait for the reception of one or more signals, then trigger some processing.
+
+*Wait_For_Signals_Reactive* is a template workflow that sends a ready notification for all the signals specified in the variable SIGNALS, then waits until one signal among those specified is received.
+This workflow contains no loop. It performs a blocking wait, then immediately triggers some processing at the reception of a signal.
+
+*Wait_For_Signal_With_Variables* is a template workflow that deals signals having input parameters. It contains a single task that sends a ready notification for all the signals (with input parameters) specified in the variable SIGNALS, then waits until one signal (among those specified) is received. The signal's input parameters have predefined values that can be changed when sending the signal.
+
```
Comment (nebilbenmabrouk) [inline src/docs/user/ProActiveUserGuide.adoc:2637]: We need to apply the same changes as in the PR of proactive-examples.

**Review by nebilbenmabrouk — APPROVED** [review]

**Comment by fviale** [comment]
Retest this please

**Pull request merged**